### PR TITLE
feat: timestamp serializing and de-serializing

### PR DESCRIPTION
**Pull request by srchase**
Adds support for timestampFormat trait

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/client-kinesis-browser/model/ServiceMetadata.ts [hunk 1]
```diff
@@ -12,4 +12,4 @@ export const ServiceMetadata: _ServiceMetadata_ = {
     targetPrefix: 'Kinesis_20131202',
     uid: 'kinesis-2013-12-02'
 };
-export const clientVersion: string = '0.1.0-preview.2';
\ No newline at end of file
+export const clientVersion: string = '0.1.0-preview.3';
```
Comment (AllanFly120) [inline packages/client-kinesis-browser/model/ServiceMetadata.ts]: I'm wondering why GitHub see this as a diff. The metadata is already up-to-date in master.
https://github.com/aws/aws-sdk-js-v3/blob/master/packages/client-kinesis-browser/model/ServiceMetadata.ts#L15

Comment (AllanFly120) [reply packages/client-kinesis-browser/model/ServiceMetadata.ts]: I find many of the diffs come the last commit: https://github.com/aws/aws-sdk-js-v3/commit/51eb26f09f0621dc4bb51b2627562c6badac815d. Maybe you can rebase before open this PR?

Comment (srchase) [reply packages/client-kinesis-browser/model/ServiceMetadata.ts]: This is fixed now.

File: packages/json-parser/src/index.spec.ts [hunk 2]
```diff
@@ -126,16 +126,28 @@ describe('JsonParser', () => {
...
-        it('should return undefined if the input is not a number', () => {
+        // it('should convert rfc822 timeStamps to date objects', () => {
+        //     expect(jsonBody.parse(timestampShape, rfc822Timestamp))
+        //         .toEqual(date);
+        // });
+
+        // it('should convert iso8601 timeStamps to date objects', () => {
+        //     expect(jsonBody.parse(timestampShape, isoTimestamp))
+        //         .toEqual(date);
+        // });
```
Comment (AllanFly120) [inline packages/json-parser/src/index.spec.ts]: nit: clean the commented code

File: packages/json-parser/src/index.spec.ts [hunk 3]
```diff
@@ -126,16 +126,28 @@ describe('JsonParser', () => {
 
     describe('timestamps', () => {
         const timestampShape: Member = {shape: {type: "timestamp"}};
-        const date = new Date('2017-05-22T19:33:14.000Z');
-        const timestamp = 1495481594;
+        const unixTimestamp = 1495481594;
+        const rfc822Timestamp = 'Mon, May 22, 2017 7:33:14 GMT'
+        const isoTimestamp = '2017-05-22T19:33:14.000Z';
```
Comment (AllanFly120) [inline packages/json-parser/src/index.spec.ts]: Are these referred anywhere?

File: packages/protocol-rest/src/RestSerializer.ts [hunk 4]
```diff
@@ -134,22 +132,23 @@ export class RestSerializer<StreamType> implements
             const {
                 location,
                 locationName = memberName,
-                shape: memberShape
+                //shape: memberShape
```
Comment (AllanFly120) [inline packages/protocol-rest/src/RestSerializer.ts]: nit: remove the commented lines

File: packages/json-parser/src/index.spec.ts (hunk 3 above)
Comment (srchase) [reply packages/json-parser/src/index.spec.ts]: Used in re-enabled tests.

**Pull request merged**